### ISSUE TITLE
Generals: Experience menu missing boxes

### ISSUE THREAD

**Issue by charliefoxtwo** [OpenSAGE/OpenSAGE#862]
Not sure if this is an issue with wnd drawing or what.

### Expected
![image](https://github.com/OpenSAGE/OpenSAGE/assets/30303272/932aaeb0-946d-4d9b-9b27-aa3a9b3700a3)

### Actual
Note the missing boxes around the skill points and rank headers.
![image](https://github.com/OpenSAGE/OpenSAGE/assets/30303272/fd0a5143-1504-4029-ac06-b56ca1e807bd)

### Actual
The zero hour experience window just isn't drawing at all for some reason. It looks like Zero Hour uses images for the background here, and for the life of me I can't figure out why the image isn't appearing. Everything seems to be correct, the image is present in the window and the call is being made to draw it, but nothing seems to appear.
![image](https://github.com/OpenSAGE/OpenSAGE/assets/30303272/f6951c09-c281-43e0-b7d5-00cdcbec0204)

**Comment by c001ac1d** [comment]
hey! regarding missing text boxes:

https://github.com/OpenSAGE/OpenSAGE/blob/f3f8364885eaccd1c89c3d5f675a0a8b6f4861cb/src/OpenSage.Game/Gui/Wnd/Controls/Label.cs#L16-L19

looks like the label control is just drawing the text and is ignoring any border colors passed in EnabledDrawData?

i've made a simple test on my end:
```csharp
        public Label(WndWindowDefinition wndWindow)
        {
            TextAlignment = wndWindow.StaticTextData.Centered
                ? TextAlignment.Center
                : TextAlignment.Leading;

            if (wndWindow.EnabledDrawData.Items[0] != null)
            {
                BorderColor = wndWindow.EnabledDrawData.Items[0].BorderColor.ToColorRgbaF();
            }
            else
            {
                BorderColor = ColorRgbaF.Transparent;
            }
           
            BorderWidth = 1;
        }
```

```csharp
        protected override void DrawOverride(DrawingContext2D drawingContext)
        {
            DrawText(drawingContext, TextAlignment);
            drawingContext.DrawRectangle(ClientRectangle.ToRectangleF(), BorderColor, BorderWidth);
        }
```
![image](https://github.com/OpenSAGE/OpenSAGE/assets/147560985/3be0ba77-a8ad-49cc-8dd3-24fb59b3a330)

looks nice now

**Comment by charliefoxtwo** [comment]
Nice find!

**Comment by c001ac1d** [comment]
so, here's what i've been able to find about the ZH problem so far:

1. the `MappedImage` that the engine is failing to render is called `GeneralsPowerMenu_China`. you can find its definition in the `INIZH.big` archive at `Data\INI\MappedImages\TextureSize_512\SCPurchasePowers512.INI`
```
MappedImage GeneralsPowerMenu_China
  Texture = SCPurchasePowers512_003.tga
  TextureWidth = 512
  TextureHeight = 512
  Coords = Left:1 Top:1 Right:393 Bottom:431
  Status = NONE
End
```

2. the texture file is in the `TexturesZH.big` archive at `Art\Textures\scpurchasepowers512_003.dds` 
3. if you inspect the processed texture of the `BackgroundImage` in RenderDoc, you'll see that it's actuallly just fully transparent lol
![image](https://github.com/OpenSAGE/OpenSAGE/assets/147560985/12e1cb7a-057e-4594-8ccb-fb22bc4654c3)

in summary, we have a DDS texture that we have failed to copy pixels from, i think it happens at this stage:
https://github.com/OpenSAGE/OpenSAGE/blob/f3f8364885eaccd1c89c3d5f675a0a8b6f4861cb/src/OpenSage.Game/Gui/Wnd/Images/MappedImageUtility.cs#L11-L52

**Comment by charliefoxtwo** [comment]
We can split the ZH image rendering into a separate issue as well if that would make more sense.

Any idea why we would fail to copy the pixels? I'm not really familiar with any of the GUI code.

**Comment by c001ac1d** [comment]
yeah, a separate issue would be better
no idea why the problem occurs, I'll look further into it tommorow

**Comment by charliefoxtwo** [comment]
https://github.com/OpenSAGE/OpenSAGE/issues/907 created to track the ZH issue
